plugins/google/compute/insDefaultServiceAccount.js

```diff
@@ -0,0 +1,88 @@
+        var defaultServiceAccount = project.data[0].defaultServiceAccount;
+
+        async.each(regions.instances.compute, (region, rcb) => {
+            var zones = regions.zones;
+            var myError = {};
+            var noInstances = {};
+            var defaultServiceAccountInstances = [];
+            var validServiceAccountInstances = [];
+            async.each(zones[region], function(zone, zcb) {
+                var instances = helpers.addSource(cache, source,
+                    ['instances', 'compute','list', zone]);
+
+                if (!instances) return zcb();
+
+                if (instances.err || !instances.data) {
+                    if (!myError[region]) {
+                        myError[region] = [];
+                    }
+                    myError[region].push(zone);
+                    return zcb();
+                }
+
+                if (!instances.data.length) {
+                    if (!noInstances[region]) {
+                        noInstances[region] = [];
+                    }
+                    noInstances[region].push(zone);
+                    return zcb();
+                }
+
+                instances.data.forEach(instance => {
+                    if (instance.serviceAccounts &&
+                        instance.serviceAccounts.length) {
+                        var found = instance.serviceAccounts.find(account => account.email == defaultServiceAccount);
+                        if (found) {
+                            defaultServiceAccountInstances.push(instance.id)
+                        }
+                        else {
+                            validServiceAccountInstances.push(instance.id)
+                        }
+                    } else {
+                        validServiceAccountInstances.push(instance.id)
+                    }
+                });
+            });
+
+            if (myError[region] &&
+                zones[region] &&
+                (myError[region].join(',') === zones[region].join(','))) {
+                helpers.addResult(results, 3, 'Unable to query instances' , region);
+            } if (noInstances[region] &&
+                zones[region] &&
+                (noInstances[region].join(',') === zones[region].join(','))) {
+                helpers.addResult(results, 0, 'No instances found in the region' , region);
+            } if (defaultServiceAccountInstances.length) {
+                var myInstanceStr = defaultServiceAccountInstances.join(", ");
+                helpers.addResult(results, 2,
+                    `Default service account is used for following instances: ${myInstanceStr}`, region);
+            } if (validServiceAccountInstances.length) {
+                var myInstanceStr = validServiceAccountInstances.join(", ");
+                helpers.addResult(results, 0,
+                    `Default service account is not used for following instances: ${myInstanceStr}`, region);
+            }
```

Comment:
are these supposed to be else if's?

Reply:
There can be multiple instances in one zone. And multiple instances can have different settings. That's why multiple conditions can be fulfilled at the same iteration.